### PR TITLE
Allow hiding chat user list

### LINKED ISSUE #1211: Allow hiding chat user list
On small clients, the user list takes up a lot of space, ads an option to hide it.

http://forums.faforever.com/viewtopic.php?f=2&t=17068#p170815

### PR DESCRIPTION
**Pull request by BlackYps**
Fixes #1211 
grafik

### CODE REVIEW

File: src/main/java/com/faforever/client/chat/ChannelTabController.java [hunk 1]
```diff
@@ -661,4 +662,16 @@ public void onPlayerOnline(PlayerOnlineEvent event) {
...
+  public void toggleSidePane(ActionEvent actionEvent) {
+    if (channelTabScrollPaneVBox.isManaged()) {
+      toggleSidePaneButton.setText("");
+      channelTabScrollPaneVBox.setManaged(false);
+      splitPane.setDividerPositions(1.0);
+    } else {
+      toggleSidePaneButton.setText("");
+      channelTabScrollPaneVBox.setManaged(true);
+      splitPane.setDividerPositions(0.8);
+    }
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChannelTabController.java]: Maybe look into Bindings could be an example to learn them here

File: src/test/java/com/faforever/client/chat/ChannelTabControllerTest.java [hunk 2]
```diff
@@ -389,4 +389,12 @@ public void testChannelTopicUpdate() {
     WaitForAsyncUtils.waitForFxEvents();
     assertEquals(instance.topicText.getChildren().size(), 4);
   }
+
+  @Test
+  public void testChannelListHide() {
+    instance.toggleSidePaneButton.fire();
+    WaitForAsyncUtils.waitForFxEvents();
+    assertFalse(instance.channelTabScrollPaneVBox.isManaged());
+    assertEquals(instance.toggleSidePaneButton.getText(), "");
```
Comment (1-alex98) [inline src/test/java/com/faforever/client/chat/ChannelTabControllerTest.java:399]: Test show as well ...

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/chat/ChannelTabController.java (hunk 1 above)
Comment (BlackYps) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: I think the bind method only works when I want to copy a property value Do you mean using a changeListener?

Comment (1-alex98) [reply src/main/java/com/faforever/client/chat/ChannelTabController.java]: Binds.createStringBinding allows for custom Bindings

**Comment by 1-alex98** [comment]
I fear we need to redo this after @micheljung PR is merged which i am gonna do shortly. See the wiki on how adding icons works after the merge of the PR

**Comment by 1-alex98** [comment]
@Sheikah45 I'd love your review

**Review by 1-alex98 — APPROVED** [review]
tested it works also its my code so its awesome

**Comment by Sheikah45** [comment]
Needs to be rebased to develop

File: src/test/java/com/faforever/client/chat/ChannelTabControllerTest.java (hunk 2 above)
Comment (Sheikah45) [reply src/test/java/com/faforever/client/chat/ChannelTabControllerTest.java:399]: Show test is still needed yes?

**Review by Sheikah45 — CHANGES_REQUESTED** [review]
Looks good and worked when tested but CI fails

**Pull request closed without merging**